src/NuGet.Core/NuGet.ProjectModel/ProjectRestoreMetadata.cs

```diff
@@ -50,6 +50,9 @@ public class ProjectRestoreMetadata : IEquatable<ProjectRestoreMetadata>
         /// </summary>
         public string PackagesPath { get; set; }
 
+        public string CacheFilePath {
+            get { return OutputPath + "project.nuget.json"; }  }
```

Comment:
The folder could have multiple projects, so it needs to match the project.assets.json name or include the project name.